app/src/main/java/lt/vilnius/tvarkau/utils/TextUtils.kt

```diff
@@ -0,0 +1,26 @@
+package lt.vilnius.tvarkau.utils
+
+import java.util.*
+import java.util.regex.Pattern
+import java.util.regex.Pattern.CASE_INSENSITIVE
+
+/**
+ * @author Martynas Jurkus
+ */
+
+object TextUtils {
+
+    private val PROBLEM_ID_REGEXP = "E50-\\S+?\\)"
+    private val PROBLEM_ID_PATTERN = Pattern.compile(PROBLEM_ID_REGEXP, CASE_INSENSITIVE)
+
+    fun findProblemIdOccurrences(source: String): List<String> {
+        val result = ArrayList<String>()
+        val matcher = PROBLEM_ID_PATTERN.matcher(source)
+
+        while (matcher.find()) {
+            result.add(matcher.group())
+        }
+
+        return result
```

Comment:
I think it can be simplified to (at least tests are passing)
```kotlin
    private val PROBLEM_ID_PATTERN = "[A-Z]\\d{2}-\\d+/\\d+\\(\\S+?\\)".toRegex(RegexOption.IGNORE_CASE)

    fun findProblemIdOccurrences(source: String): List<String> =
            PROBLEM_ID_PATTERN.findAll(source).map { s -> s.value }.toList()
```